harness/determined/keras/_tf_keras_trial.py

```diff
@@ -760,6 +758,18 @@ def _launch_evaluate(self) -> Any:
         )
         logging.debug(f"Worker finished model.evaluate() with metrics: {metrics_values}.")
 
+        # Clean up the enqueuer if we started one.
+        if isinstance(self.validation_data, tf.keras.utils.Sequence):
+            enqueuer.stop()
+            self.enqueuers.remove(enqueuer)
+
+            # A special side-effect of converting the keras sequence to a generator and passing
+            # steps explicitly is that keras will exit our generator after N steps and the
+            # Sequence.on_epoch_end() that normally runs after the last yield won't run at all
+            # because the fit loop will call next() exactly `steps` times.  So we try to match the
+            # exact keras behavior by manually calling on_epoch_end() here.
+            self.validation_data.on_epoch_end()
```

Comment:
Can there ever be a case where calling this here is wrong? e.g., users call configure_fit w/ steps_per_epoch and validation_steps specified, such that partial epoch or multiple epochs are actually validated per validation workload?  (it seems fine if so but it's worth public documentation if there are any corner cases)

Reply:
I don't think that's possible.

This is the signature of configure_fit:
```python
    def configure_fit(
        self,
        verbose: Optional[bool] = None,
        class_weight: Any = _arg_not_provided,
        workers: Optional[int] = None,
        use_multiprocessing: Optional[bool] = None,
        max_queue_size: Optional[bool] = None,
        shuffle: Optional[bool] = None,
        validation_steps: Any = _arg_not_provided,
    ) -> None:
```
and `validation_steps` only applies to tf.data.Dataset objects (per the keras standard behavior).